### ISSUE TITLE
Integrated document scanning

### ISSUE THREAD

**Issue by bobleeSwagger2** [nextcloud/android#4435]
Hi. Please add document scanning to the app to bring it on par with the iOS counterpart. This would prevent having to use insecure third party apps to scan, then upload to NC. Thanks

**Comment by articlecat** [comment]
Yes, an open source document scanning feature would be very useful.

Right now, most Android users who need this feature are using proprietary apps, which are less trustworthy. Last month, CamScanner (which has over 100 million Android downloads) was [discovered to have included malware](https://www.zdnet.com/article/android-google-play-app-with-100-million-downloads-starts-to-deliver-malware).

**Comment by SeeSpotRun** [comment]
Perhaps the relevant [code](https://github.com/ctodobom/OpenNoteScanner/blob/master/app/src/main/java/com/todobom/opennotescanner/ImageProcessor.java) from the FOSS app `Open Note Scanner` can be used as a starting point?

**Comment by Happyfeet01** [comment]
Would be very useful. I switched from iOS back to android and miss this feature.

**Comment by benjamindahl** [comment]
I still don't see the scan feature. Is it on the way?

**Comment by EVOTk** [comment]
Hello, 
Available in the new version still in RC

On the other hand, it does not work for me, its use, when validating the upload of the document, it loads endlessly then crashes the application.

https://github.com/nextcloud/android/releases/tag/rc-3.20.0-01

![Screenshot_20220329_165153_com nextcloud client](https://user-images.githubusercontent.com/45015615/160640454-90a34f68-ae43-4a49-a7d2-6ed702e26a35.jpg)

```### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Bitmap.createBitmap(this…th, this.height, m, true) must not be null
    at com.zynksoftware.documentscanner.common.extensions.BitmapExtensionsKt.scaledBitmap(BitmapExtensions.kt:50)
    at com.zynksoftware.documentscanner.ui.imagecrop.ImageCropFragment.initializeCropping(ImageCropFragment.kt:98)
    at com.zynksoftware.documentscanner.ui.imagecrop.ImageCropFragment.access$initializeCropping(ImageCropFragment.kt:45)
    at com.zynksoftware.documentscanner.ui.imagecrop.ImageCropFragment$onViewCreated$2.run(ImageCropFragment.kt:77)
    at android.os.Handler.handleCallback(Handler.java:900)
    at android.os.Handler.dispatchMessage(Handler.java:103)
    at android.os.Looper.loop(Looper.java:219)
    at android.app.ActivityThread.main(ActivityThread.java:8393)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30200051`
* Build flavor: `generic`

### Device information
* Brand: `HONOR`
* Device: `HWYAL`
* Model: `YAL-L21`
* Id: `HUAWEIYAL-L21`
* Product: `YAL-L21EEA`

### Firmware
* SDK: `29`
* Release: `10`
* Incremental: `10.1.0.241C431`
```

**Comment by AlvaroBrey** [comment]
Implemented in #9717 . I'll open a different issue for your crash @EVOTk
